### PR TITLE
(feat) add connection backlog support

### PR DESCRIPTION
**Pull request by vinayakhulawale**

### CODE REVIEW

**Comment by vinayakhulawale** [comment]
@agentzh or @thibaultcha - any interest in merging this ?

**Comment by rainingmaster** [comment]
@vinayakhulawale 
Hello, could you help to add some test case to cover the new code?

File: t/sanity.t [hunk 1]
```diff
@@ -1651,3 +1651,41 @@ success
 --- no_error_log
 [error]
 --- timeout: 20
+
+=== TEST 24: connected with no error when pool opts are provided
+--- http_config eval: $::HttpConfig
+--- config
+    location /t {
+        content_by_lua '
```
Comment (doujiang24) [inline t/sanity.t:1659]: it's better use `content_by_lua_block` for the new tests.

**Comment by doujiang24** [comment]
@vinayakhulawale thanks, already merged in https://github.com/openresty/lua-resty-mysql/pull/106

**Comment by vinayakhulawale** [comment]
Thanks @doujiang24 !

**Pull request closed without merging**